examples/rosenbrock.py

```diff
@@ -45,87 +32,50 @@ def sensfunc(xdict, funcsDict):
     return funcsSens, fail
 
 
-# Matrix-free algorithm return functions
-def objgrad(xdict):
-    x = xdict["xvars"]
-
-    objsens = {}
-    objsens["obj"] = {
-        "xvars": np.array([2 * 100 * (x[1] - x[0] ** 2) * (-2 * x[0]) - 2 * (1 - x[0]), 2 * 100 * (x[1] - x[0] ** 2)])
-    }
-    fail = False
-
-    return objsens, fail
-
-
-def jprod(xdict, pdict, sparse_only):
-    x = xdict["xvars"]
-    p = pdict["xvars"]
-    qdict = {}
-
-    if constrained:
-        gcon = np.array([-3 * (x[0] - 1) ** 2, -1])
-        qdict["con"] = np.array([np.dot(gcon, p)])
-    fail = False
-
-    return qdict, fail
-
-
-def jtprod(xdict, qdict, sparse_only):
-    x = xdict["xvars"]
-
-    pdict = {}
-    if constrained:
-        q = qdict["con"]
-        gcon = np.array([-3 * (x[0] - 1) ** 2, -1])
-        pdict["xvars"] = q * gcon
-    else:
-        pdict["xvars"] = np.zeros(len(x))
-    fail = False
-
-    return pdict, fail
-
-
-if sens == "none":
-    sens = None
-if sens == "user":
+if args.sens == "user":
     sens = sensfunc
-if sens == "matrix-free":
-    sens = [objgrad, jprod, jtprod]
+else:
+    sens = args.sens
 
 # Instantiate Optimization Problem
 optProb = Optimization("Rosenbrock function", objfunc)
 optProb.addVarGroup("xvars", 2, "c", value=[3, -3], lower=-5.12, upper=5.12, scale=[1.0, 1.0])
-optProb.finalizeDesignVariables()
-if constrained:
+if args.constrained:
     optProb.addCon("con", upper=0, scale=1.0)
 optProb.addObj("obj")
 
 # Create optimizer
 opt = OPT(args.opt, options=optOptions)
-if testHist == "no":
-    # Just run a normal run
-    sol = opt(optProb, sens=sens, sensMode=sensMode)
-    # print(sol.fStar)
-    print(sol)
-else:
-    # First call just does 10 iterations
-    if args.opt.lower() == "snopt":
-        opt.setOption("Major iterations limit", 10)
-        solSnopt1 = opt(optProb, sens=sens, sensMode="pgc", storeHistory="opt_hist")
 
-        # Now we are allowed to do 50
-        opt.setOption("Major iterations limit", 50)
-        if testHist == "hot":
-            solSnopt2 = opt(optProb, sens=sens, sensMode=sensMode, hotStart="opt_hist", storeHistory="opt_hist")
-        else:
-            solSnopt2 = opt(optProb, sens=sens, sensMode=sensMode, coldStart="opt_hist", storeHistory="opt_hist")
+if args.testHotstart:
```

Comment:
Is this the best practice to map a specific option to different optimizers? What if we are dealing with multiple options and not just `max_iter`?

Reply:
Yeah this is a good point. Maybe we should restrict this to run with SLSQP only, and then everything would be simpler.

Reply:
I've updated the rosenbrock example to only run SLSQP.